### PR TITLE
Improve documentation for cloudflare_access_group resource

### LINKED ISSUE #2299: cloudflare_access_group documentation is not clear for azure
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

```
➜ terraform -v            
Terraform v1.4.2
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v4.1.0

```

### Affected resource(s)

`cloudflare_access_group`

image

### Terraform configuration files

```hcl

resource "cloudflare_access_group" "g_test" {
  name = "g_test"
  zone_id = "<redacted>"
  include {
    azure {
      identity_provider_id = "<redacted>"
      id = ["<redacted>"]
    }
  }
}
```

### Link to debug output

na

### Panic output

_No response_

### Expected output

The documentation states that under `include.azure` the field `id` is **read only** when it is in fact not

### Actual output

The documentation should read more like

**Nested Schema for [`include.azure`](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/access_group#nested-schema-for-includeazure)**

### Required

* [`identity_provider_id`](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/access_group#identity_provider_id) (string) - The ID of the Azure Identity provider
* [id](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/access_group#id) (List of String) The ID of the Azure group or user.

### Steps to reproduce

Look at documentation https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/access_group#nestedblock--include--azure

### Additional factoids

The provider documentation lacks examples for all the options it has.

Would be worth while adding in an example for the azure group like:

``` tf
resource "cloudflare_access_group" "g_test" {
  name = "g_test"
  zone_id = "<redacted>"
  include {
    azure {
      identity_provider_id = "<redacted>"
      id = ["<redacted>"]
    }
  }
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by Nmishin**
Closes #2299

### CODE REVIEW

**Comment by Nmishin** [comment]
this is for https://github.com/cloudflare/terraform-provider-cloudflare/issues/2299

File: examples/resources/cloudflare_access_group/resource.tf [hunk 1]
```diff
@@ -22,3 +22,16 @@ resource "cloudflare_access_group" "test_group" {
     ip = [var.office_ip]
   }
 }
+
+# Allow members of an Azure Group. The ID is the group UUID (id) in Azure:
```
Comment (jacobbednarz) [inline examples/resources/cloudflare_access_group/resource.tf:26]: i'll 👍 this but generally, we don't have kitchen sink style examples in our Terraform documentation. this should live in our developer documentation with an overview of how to use Azure as opposed to only in our Terraform implementation.

File: .changelog/2332.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+resource/cloudflare_access_group.go: Add example of usage, add descriptions of fields.
```
Comment (jacobbednarz) [inline .changelog/2332.txt:2]:
```suggestion
resource/cloudflare_access_group: Add example of usage of Azure 
```

File: examples/resources/cloudflare_access_group/resource.tf (hunk 1 above)
Comment (Nmishin) [reply examples/resources/cloudflare_access_group/resource.tf]: got it, thank you

**Pull request merged**